packit_service/worker/handlers/testing_farm.py

```diff
@@ -94,9 +94,21 @@ def __init__(
         self._db_trigger: Optional[AbstractTriggerDbType] = None
         self._testing_farm_job_helper: Optional[TestingFarmJobHelper] = None
 
+    def check_if_contributor_can_run_job_and_report(self, actor: str) -> bool:
+        if self.job_config.metadata.use_internal_tf and not self.project.can_merge_pr(
+            actor
+        ):
+            self.testing_farm_job_helper.report_status_to_tests(
+                description=f"{actor} can't run tests internally",
```

Comment:
sounds like a good-enough solution to me, short-term

not sure if we can display text on the check page stating that the owner of the project needs to retrigger using `/packit test`

Reply:
Good point, I'll take a look if we can show more info in the run's UI.

Reply:
> short-term

@TomasTomecek that shouldn't be a short term solution...;) With this, we can easily do the same for internal Copr, or anything else. Do you have a better approach so we can do that now?

And regarding the runs UI, this will require edits in the reporter -- I'll send it as a follow-up.

Reply:
I thought that long-term we want to figure out a permission system so that upstream maintainers can grant contributors ability to run CI on their own